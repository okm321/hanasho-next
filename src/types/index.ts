export * from './header'
export * from './product'
export * from './slide'
