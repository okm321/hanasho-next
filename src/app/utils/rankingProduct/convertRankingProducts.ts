import { RANKING_PRODUCT_COLOR_MAP } from '@/constants/Product'
import { ProductAPI } from '@/types'
import {
  ConvertedRankingProduct,
  RankingProductColor,
} from '@/types/product/rankingProduct'

const getRankColor = (rank: number): RankingProductColor | 'black' => {
  if (rank === 1 || rank === 2 || rank === 3) {
    return RANKING_PRODUCT_COLOR_MAP[rank]
  }
  return 'black'
}

export const convertRankingProducts = (
  products: ProductAPI[]
): ConvertedRankingProduct[] => {
  return products.map((product, index) => {
    const rank = index + 1
    return {
      ...product,
      rankingInfo: {
        ranking: rank,
        backgroundColor: getRankColor(rank),
        textColor: 'white',
      },
    }
  })
}
