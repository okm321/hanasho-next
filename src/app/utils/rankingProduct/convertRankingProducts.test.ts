import { ProductAPI } from '@/types'
import { convertRankingProducts } from './convertRankingProducts'
import { times } from 'lodash-es'

const createRankingProduct = (num: number): ProductAPI[] => {
  return times(num, (index) => {
    return {
      id: index + 1,
      list_image: `https://example.com/image${index + 1}.jpg`,
      name: `商品${index + 1}`,
      subtitle: `サブタイトル${index + 1}`,
      price: (index + 1) * 1000,
    }
  })
}

describe('convertRankingProducts', () => {
  it('人気ランキングのデータが3個以内の場合', () => {
    expect(convertRankingProducts(createRankingProduct(2))).toMatchSnapshot()
  })
  it('人気ランキングのデータが3個以上の場合', () => {
    expect(convertRankingProducts(createRankingProduct(10))).toMatchSnapshot()
  })
})
