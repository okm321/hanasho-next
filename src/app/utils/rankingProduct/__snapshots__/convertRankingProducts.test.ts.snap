// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertRankingProducts 人気ランキングのデータが3個以上の場合 1`] = `
[
  {
    "id": 1,
    "list_image": "https://example.com/image1.jpg",
    "name": "商品1",
    "price": 1000,
    "rankingInfo": {
      "backgroundColor": "#e5d223",
      "ranking": 1,
      "textColor": "white",
    },
    "subtitle": "サブタイトル1",
  },
  {
    "id": 2,
    "list_image": "https://example.com/image2.jpg",
    "name": "商品2",
    "price": 2000,
    "rankingInfo": {
      "backgroundColor": "#b1b1b1",
      "ranking": 2,
      "textColor": "white",
    },
    "subtitle": "サブタイトル2",
  },
  {
    "id": 3,
    "list_image": "https://example.com/image3.jpg",
    "name": "商品3",
    "price": 3000,
    "rankingInfo": {
      "backgroundColor": "#b77437",
      "ranking": 3,
      "textColor": "white",
    },
    "subtitle": "サブタイトル3",
  },
  {
    "id": 4,
    "list_image": "https://example.com/image4.jpg",
    "name": "商品4",
    "price": 4000,
    "rankingInfo": {
      "backgroundColor": "black",
      "ranking": 4,
      "textColor": "white",
    },
    "subtitle": "サブタイトル4",
  },
  {
    "id": 5,
    "list_image": "https://example.com/image5.jpg",
    "name": "商品5",
    "price": 5000,
    "rankingInfo": {
      "backgroundColor": "black",
      "ranking": 5,
      "textColor": "white",
    },
    "subtitle": "サブタイトル5",
  },
  {
    "id": 6,
    "list_image": "https://example.com/image6.jpg",
    "name": "商品6",
    "price": 6000,
    "rankingInfo": {
      "backgroundColor": "black",
      "ranking": 6,
      "textColor": "white",
    },
    "subtitle": "サブタイトル6",
  },
  {
    "id": 7,
    "list_image": "https://example.com/image7.jpg",
    "name": "商品7",
    "price": 7000,
    "rankingInfo": {
      "backgroundColor": "black",
      "ranking": 7,
      "textColor": "white",
    },
    "subtitle": "サブタイトル7",
  },
  {
    "id": 8,
    "list_image": "https://example.com/image8.jpg",
    "name": "商品8",
    "price": 8000,
    "rankingInfo": {
      "backgroundColor": "black",
      "ranking": 8,
      "textColor": "white",
    },
    "subtitle": "サブタイトル8",
  },
  {
    "id": 9,
    "list_image": "https://example.com/image9.jpg",
    "name": "商品9",
    "price": 9000,
    "rankingInfo": {
      "backgroundColor": "black",
      "ranking": 9,
      "textColor": "white",
    },
    "subtitle": "サブタイトル9",
  },
  {
    "id": 10,
    "list_image": "https://example.com/image10.jpg",
    "name": "商品10",
    "price": 10000,
    "rankingInfo": {
      "backgroundColor": "black",
      "ranking": 10,
      "textColor": "white",
    },
    "subtitle": "サブタイトル10",
  },
]
`;

exports[`convertRankingProducts 人気ランキングのデータが3個以内の場合 1`] = `
[
  {
    "id": 1,
    "list_image": "https://example.com/image1.jpg",
    "name": "商品1",
    "price": 1000,
    "rankingInfo": {
      "backgroundColor": "#e5d223",
      "ranking": 1,
      "textColor": "white",
    },
    "subtitle": "サブタイトル1",
  },
  {
    "id": 2,
    "list_image": "https://example.com/image2.jpg",
    "name": "商品2",
    "price": 2000,
    "rankingInfo": {
      "backgroundColor": "#b1b1b1",
      "ranking": 2,
      "textColor": "white",
    },
    "subtitle": "サブタイトル2",
  },
]
`;
